servo/connectors/opsani_dev.py

```diff
@@ -214,6 +214,40 @@ async def check_kubernetes_service_type(self) -> None:
                 f"expected service type of ClusterIP or LoadBalancer but found {service.spec.type}"
             )
 
+    @servo.checks.require("Target container resource requests are within limits")
+    async def check_target_container_resources_within_limits(self) -> None:
+        # Load the Deployment
+        deployment = await servo.connectors.kubernetes.Deployment.read(
+            self.config.deployment,
+            self.config.namespace
+        )
+        # assert deployment, f"failed to read deployment '{self.config.deployment}' in namespace '{self.config.namespace}'"
+
+        # Find the target Container
+        target_container = None
+        for container in deployment.containers:
+            if container.name == self.config.container:
+                target_container = container
+                break
+
+        assert target_container, f"failed to find container '{self.config.container}' when verifying resource limits"
+
+        # Get resource requirements from container
+        container_cpu_request = servo.connectors.kubernetes.Millicore.parse(target_container.resources.requests["cpu"])
+        container_memory_request = servo.connectors.kubernetes.ShortByteSize.validate(target_container.resources.requests["memory"])
```

Comment:
You can turn these into `@property` methods on the `Container` class so that the code reads `target_container.cpu` and `target_container.memory`

Reply:
Struggling with this. The concept is easy, but I am getting stuck in figure out the data types. This may be more of an IDE / visibility issue for me. May need help with the tooling.

Reply:
@lswank Here's the python [client model for ResourceRequirements](https://github.com/kubernetes-client/python/blob/master/kubernetes/docs/V1ResourceRequirements.md). As you can see, requests/limits are both dict[str, str] so I'm thinking your return type should be a string

Also @blakewatters shouldn't we also add property decorators `target_container.cpu_limit` and `target_container.memory_limit` for the sake of completeness?